data_steward/validation/participants/identity_match.py

```diff
@@ -1,379 +1,336 @@
+import validation.participants.readers as readers
+import validation.participants.writers as writers
 
-
-def _get_observation_match_values(dataset):
-    """
-    Get the desired matching values from the combined observation table.
-
-    This retrieves all possible matching values from the observation table
-    used in participant matching.  Returned query data is limited to the
-    person_id, observation_concept_id, and the value_as_string fields.
-
-    :param dataset:  The string name of the dataset to use when matching participants
-
-    :return:  A dictionary with observation_concept_id as keys.  Each value is
-        a dictionary of person_ids with the associated value of the concept_id.
-        For example:
-        { 1585260: {person_id_1:  "email_address", person_id_2: "email_address"},
-          1585596: {person_id_1: "first_name", person_id_2: "first_name"},
-          1585597: {person_id_1: "middle_name", },
-          1585598: {person_id_1: "last_name", person_id_2: "last_name"},
-          ...,
-        }
-    """
-    query_string = consts.PPI_OBSERVATION_VALUES.format(dataset=dataset,
-                                                        table='observation')
-    print(query_string)
-    #    results = bq_utils.query(query_string)
-    return {}
-
-
-def _get_hpo_site_names():
-    # This will have to do until I can talk with other engineers.
-    return []
+LOGGER = logging.getLogger(__name__)
 
 
-def _get_pii_names(hpo):
+def _compare_address_lists(list_one, list_two):
     """
-    Get names from the site's PII table.
-
-    :param hpo: hpo string to use when identifying table names for lookup.
+    Counts the number of elements in list one that are not in list two.
 
-    :return:  A list of tuples with the first tuple element as the pserson_id
-    and the second tuple element as the first name, the third tuple element
-    is the middle name, the fourth tuple element is the last name.  For example,
-    [(1, 'suzie', '', 'q'), (48, 'johnny', 'B', 'Good'), (99, 'roy', 'g', 'biv')]
+    :param list_one: list of strings to check for existence in list_two
+    :param list_two: list of strings to use for existence check of
+        list_one parts
+    :return: the count of items in list_one that are missing from list_two
     """
-    return []
+    diff = 0
+    for part in list_one:
+        if part not in list_two:
+            diff += 1
+    return diff
 
 
-def _get_pii_emails(hpo):
+def _compare_name_fields(
+        project,
+        rdr_dataset,
+        pii_dataset,
+        hpo,
+        concept_id,
+        pii_field
+    ):
     """
-    Get email addresses from the site's PII table.
+    For an hpo, compare all first, middle, and last name fields to omop settings.
 
-    :param hpo:  hpo string to use when identifying table names for lookup.
+    This compares a site's name field values found in their uploaded PII
+    tables with the values in the OMOP observation table.
```

Comment:
Doc string should be updated with the  project, rdr_dataset, pii_dataset, concept_id, pii_field parameters descriptions.